app/assets/javascripts/discourse/tests/helpers/qunit-helpers.js

```diff
@@ -248,7 +246,45 @@ export function acceptance(name, options) {
       // We do this after reset so that the willClearRender will have already fired
       resetWidgetCleanCallbacks();
     },
-  });
+  };
+
+  const needs = {
+    authenticatedUser() {
+      loggedIn = true;
+    },
+    pretender(fn) {
+      addPretenderCallback(name, fn);
+    },
+    site(changes) {
+      siteChanges = changes;
+    },
+    settings(changes) {
+      settingChanges = changes;
+    },
+  };
```

Comment:
Would it be possible to export those helper functions instead?
That would more closely align our tests suite with the latest qunit/ember testing idioms (for example: [`setupApplicationTest`](https://guides.emberjs.com/release/testing/testing-application/) or [`setupMirage`](https://www.ember-cli-mirage.com/docs/testing/acceptance-tests))

Reply:
Ember doesn't exactly do that. You get a `hooks` object from `module` which is the qunit context. That's then passed on like  `setupApplicationTest(hooks)`.

If you want to add a beforeEach/afterEach you need to call it on the hooks object,

`hooks.beforeEach(() => my code)`

And that's the pattern I stuck with, because the context object you are passed (`needs`) is the context of the test itself, not a helper that uses that object to set up the test. Does that make sense?